x-pack/plugins/security_solution/public/management/pages/policy/view/policy_advanced.tsx

```diff
@@ -15,13 +15,42 @@ import { AdvancedPolicySchema } from '../models/advanced_policy_schema';
 
 function setValue(obj: Record<string, unknown>, value: string, path: string[]) {
   let newPolicyConfig = obj;
+
+  // First set the value.
   for (let i = 0; i < path.length - 1; i++) {
     if (!newPolicyConfig[path[i]]) {
       newPolicyConfig[path[i]] = {} as Record<string, unknown>;
     }
     newPolicyConfig = newPolicyConfig[path[i]] as Record<string, unknown>;
   }
   newPolicyConfig[path[path.length - 1]] = value;
+
+  // Then, if the user is deleting the value, then we need to ensure we clean up the config.
+  // We delete any sections are the empty, whether that be an empty string,  empty object, or undefined.
+  if (value === '' || value === undefined) {
+    newPolicyConfig = obj;
+    for (let k = path.length; k >= 0; k--) {
+      const nextPath = path.slice(0, k);
+      for (let i = 0; i < nextPath.length - 1; i++) {
+        // Traverse and find the next section
+        newPolicyConfig = newPolicyConfig[nextPath[i]] as Record<string, unknown>;
+      }
+      if (
+        newPolicyConfig[nextPath[nextPath.length - 1]] === undefined ||
+        newPolicyConfig[nextPath[nextPath.length - 1]] === '' ||
+        Object.keys(newPolicyConfig[nextPath[nextPath.length - 1]] as object).length === 0
+      ) {
+        if (nextPath[nextPath.length - 1] === 'advanced') {
+          newPolicyConfig[nextPath[nextPath.length - 1]] = undefined;
```

Comment:
If I'm reading this right, it seems that if the key name is `advanced` we keep it, but set it to `undefined`, but any other key we delete it. Should the `advanced` key also be deleted? Or does that happen when you walk the `path` back to the prior level?

Reply:
the UI looks for an `advanced` key which will be initialized to `undefined` to start making decisions on how to render the form.  This is essentially resetting if it's completely emptied out

Reply:
i'm also a little confused about the if/else logic here; maybe more comments would be clarifying

Reply:
added more comments